docs/admin/templates/extending-templates/devcontainers.md

```diff
@@ -1,126 +1,345 @@
+        module "nodejs" {
+          count    = data.coder_workspace.me.start_count
+          source   = "dev.registry.coder.com/modules/nodejs/coder"
+          agent_id = coder_agent.main.id
+        }
+        ```
+
+    1. In the `devcontainers-cli` module block, add:
+
+       ```terraform
+       depends_on       = [module.nodejs]
+       ```
+
+   </details>
+
 ## Install the Dev Containers CLI
 
 Use the
 [devcontainers-cli](https://registry.coder.com/modules/devcontainers-cli) module
-to ensure the `@devcontainers/cli` is installed in your workspace:
+to install `@devcontainers/cli` in your workspace:
 
 ```terraform
 module "devcontainers-cli" {
   count    = data.coder_workspace.me.start_count
   source   = "dev.registry.coder.com/modules/devcontainers-cli/coder"
-  agent_id = coder_agent.dev.id
+  agent_id = coder_agent.main.id
 }
 ```
 
-Alternatively, install the devcontainer CLI manually in your base image.
+Alternatively, install the devcontainer CLI manually in your base image:
 
-## Configure Automatic Dev Container Startup
+```shell
+RUN npm install -g @devcontainers/cli
+```
 
-The
-[`coder_devcontainer`](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/devcontainer)
-resource automatically starts a dev container in your workspace, ensuring it's
-ready when you access the workspace:
+## Define the Dev Container Resource
+
+If you don't use [`git_clone`](#clone-the-repository), point the resource at the folder that contains `devcontainer.json`:
 
 ```terraform
-resource "coder_devcontainer" "my-repository" {
+resource "coder_devcontainer" "project" {
   count            = data.coder_workspace.me.start_count
-  agent_id         = coder_agent.dev.id
-  workspace_folder = "/home/coder/my-repository"
+  agent_id         = coder_agent.main.id
+  workspace_folder = "/home/coder/project"
 }
 ```
 
-> [!NOTE]
->
-> The `workspace_folder` attribute must specify the location of the dev
-> container's workspace and should point to a valid project folder containing a
-> `devcontainer.json` file.
+## Clone the Repository
+
+This step is optional, but it ensures that the project is present before the dev container starts.
 
-<!-- nolint:MD028/no-blanks-blockquote -->
+Note that if you use the `git_clone` module, place it before the `coder_devcontainer` resource
+and update or replace that resource to point at `/home/coder/project/${module.git_clone[0].folder_name}` so that it is only defined once:
 
-> [!TIP]
->
-> Consider using the [`git-clone`](https://registry.coder.com/modules/git-clone)
-> module to ensure your repository is cloned into the workspace folder and ready
-> for automatic startup.
+```terraform
+module "git_clone" {
+  count    = data.coder_workspace.me.start_count
+  source   = "dev.registry.coder.com/modules/git-clone/coder"
+  agent_id = coder_agent.main.id
+  url      = "https://github.com/example/project.git"
+  base_dir = "/home/coder/project"
+}
+
+resource "coder_devcontainer" "project" {
```

Comment:
It's unfortunate `folder_name` can't be used here. 😔 I'm really of two-minds using this for the sub-agent name when present but it is what it is.

We support `devcontainer.json` customization to change the name on a per-project basis.

Reply:
we don't really have that documented, but I think we should def add it in the next iteration

Reply:
adding: it would be great as part of or in addition to the `Agent naming` section